tests/php/Control/RequestHandlerTest.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace SilverStripe\Control\Tests;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\RequestHandler;
+use SilverStripe\Dev\SapphireTest;
+
+/**
+ * Tests for the RequestHandler class
+ */
+class RequestHandlerTest extends SapphireTest
+{
+    protected $usesDatabase = false;
+
+    public function provideTestLink(): array
+    {
+        return [
+            // If there's no url segment, there's no link
+            [
+                'urlSegment' => null,
+                'action' => null,
+                'expected' => null,
+            ],
+            [
+                'urlSegment' => null,
+                'action' => 'some-action',
+                'expected' => null,
+            ],
+            // The action is just addeed on after the url segment
+            [
+                'urlSegment' => 'my-controller',
+                'action' => null,
+                'expected' => 'my-controller',
+            ],
+            [
+                'urlSegment' => 'my-controller',
+                'action' => 'some-action',
+                'expected' => 'my-controller/some-action',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideTestLink
+     */
+    public function testLink(?string $urlSegment, ?string $action, ?string $expected)
+    {
+        if ($urlSegment === null) {
+            $this->expectWarning();
+            $this->expectWarningMessage('Request handler SilverStripe\Control\RequestHandler does not have a url_segment defined. Relying on this link may be an application error');
```

Comment:
These are deprecated and will be removed in PHPUnit 10 - but I think it's worth having them here while they work. We've got this sort of thing in a few places and eventually we'll have to think about what to do with it, but PHPUnit has a history of long support for not-current major releases so I suspect we won't be worrying about that until CMS 6

Reply:
They're showing up as deprecation warnings as CI. https://github.com/silverstripe/silverstripe-framework/actions/runs/4748341443/jobs/8434430162?pr=10749#step:12:135 We shouldn't be adding tech debt in new PRs. At some point we'll remove the other warnings that are currently there

There is no migration path https://github.com/sebastianbergmann/phpunit/issues/5062#issuecomment-1484644506

Can you please remove these assertions

Reply:
For the reasons I've expressed above, I think it's worth keeping these here - we probably need another opinion at this stage.
@silverstripe/core-team any thoughts?

Reply:
Note: If I remove the assertions I also _can't_ keep the test scenario where the returned value should be null, because phpunit converts the warning into an exception and the test fails.

Reply:
Maybe create a card to reach some consensus on how to handle warnings going forward. Those new PHPUnit warning are annoying enough that  I think we should aim to address them now rather than wait for PHPUnit 10.

On the narrow point of what we should do with this PR, I think it makes sense to use the "old" way and expect a warning ... just in case we decide that we really like those "expect warning" clause in the follow up card and hack them back in somehow.

Reply:
Issue created: https://github.com/silverstripe/.github/issues/40